tests/integration/api/test_eda_credential.py

```diff
@@ -306,8 +310,73 @@ def test_partial_update_eda_credential(
     }
 
 
+@pytest.mark.parametrize(
+    ("credential_type", "inputs"),
+    [
+        (enums.DefaultCredentialType.VAULT, {}),
+        (
+            enums.DefaultCredentialType.VAULT,
+            {"vault_password": "new_password"},
+        ),
+        (enums.DefaultCredentialType.AAP, {}),
+        (
+            enums.DefaultCredentialType.AAP,
+            {"host": "new_host", "username": "new user"},
+        ),
+        (enums.DefaultCredentialType.GPG, {}),
+        (enums.DefaultCredentialType.GPG, {"gpg_public_key": "new key"}),
```

Comment:
@hsong-rh this might fail as soon as we add validation for gpg key

Reply:
@mkanoor Yeah, will change it when we implement GPG key validation